CAS SARIF SAST data ingestion (#1144)
hlxsites/prisma-cloud-docs: docs/en/enterprise-edition/content-collections/application-security/get-started/connect-code-and-build-providers/third-party-ingestion/sarif-ingestion.adoc

[.task]
== Ingest SARIF Data

//You can import and consolidate findings from Static Application Security Testing (SAST) vendors that support SARIF output. This enables you to view all your Application Security (AppSec) vulnerabilities on a single platform, whether they are from third-party sources or native tools.

> Upload SAST data from third-party tools (for vendors that support SARIF output) to the Prisma Cloud platform using SARIF (Static Analysis Results Interchange Format) to analyze and remediate SAST issues directly within the platform. This feature is particularly useful for vendors that are not directly supported by Prisma Cloud. When SARIF data findings are uploaded, they are parsed to generate CWE weaknesses. 

Reviewer: ```suggestion
Upload SAST data from third-party tools (for vendors that support SARIF output) to the Prisma Cloud platform using SARIF (Static Analysis Results Interchange Format) to analyze and remediate SAST issues directly within the platform. This feature is particularly useful for vendors that are not directly supported by Prisma Cloud. When SARIF data findings are uploaded, they are parsed to generate code weaknesses. 
```

+
** *GRBAC (Generic Role-Based Access Control)*: Viewing will require permissions for Projects > View and access to the selected repositories. Ingesting SARIF files will require permissions for Providers > Create or Update. Deleting old SARIF files will require permissions for Providers > Delete 
* Onboard the repository into the system before SARIF findings for that repository can be uploaded
* Only upload findings relevant to the repository 
* Ensure that uploaded findings follow the standard SARIF specifications
* Add the Prisma Cloud IP addresses and hostname for Application Security to an xref:../../../../get-started/console-prerequisites.adoc[allow list] to enable access to the Prisma Cloud console

=== Ingest SARIF via the console
[.procedure]

. Export the required SARIF file from your third-party vendor to your host machine.

. In Application Security, select *Repositories* > click on a repository > *Add SARIF Findings*.

. Browse for and select the SARIF file from the displayed dialog box.
+

> The file is uploaded and the ingestion process begins (up to ten minutes depending on the size of the ingested data). Popups indicate if ingestion is successful. 

Reviewer: ```suggestion
The file is uploaded and the ingestion process begins (up to ten minutes depending on the size of the ingested data). Popups will notify you that the ingestion upload was successful and that the ingestion processing has beghin.
```

We need to let users know the popup is just saying the upload was successful, not that the ingestion is complete.

=== Ingest SARIF via the API

> You can ingest SARIF files via the API, enabling automation, such as in a CI/CD pipeline. Ingestion via API requires a two-stage process: first, obtain a presigned URL, and then upload the file to the provided URL.

Reviewer: ```suggestion
You can ingest SARIF files via the API, enabling automation, such as in a CI/CD pipeline.
```

There's not two steps for Prisma Cloud, only Cortex.

> [.procedure]

Reviewer: It's just 1 step, so probably not a procedure, right?

Author: Ok

> . Obtain a presigned URL.

Reviewer: ```suggestion
. Upload a file via the API.
```

+
[source,curl]
----
curl -X POST "{Prisma Cloud API URL}/code/api/v2/uploads/uploadSarifFile" \
     -H "Authorization: {Prisma Cloud generated token}" \
     -F "file=@{path/to/file.sarif};type=application/octet-stream" \
     -F "repositoryId={Repository ID}" \
     -F "fileName={file name}"
----
+
*Command properties breakdown*:
+
* *Prisma Cloud API URL*: Use your tenant API URL (for example, "https://api3.prismacloud.io"). For more information, refer to xref:../../../../get-started/console-prerequisites.adoc[Enable Access to the Prisma Cloud Console]
+
* *Prisma Cloud generated token*: Use the login endpoint to generate a JWT. Refer to https://pan.dev/prisma-cloud/api/cspm/app-login/[Prisma Cloud APIs - Login] for more information
+
* *Repository ID*: Found in the header on the *Repositories* page or the response from the List Repositories List endpoint. Refer to https://pan.dev/prisma-cloud/api/code/get-repositories/[List Repositories List] for more information

> . Upload the file to the URL.

Reviewer: Not needed
